src/System.Text.Json/src/System/Text/Json/Writer/JsonWriterState.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Text.Json
+{
+    /// <summary>
+    /// Defines an opaque type that holds and saves all the relevant state information which must be provided
+    /// to the <see cref="Utf8JsonWriter"/> to continue writing after completing a partial write.
+    /// This type is required to support reentrancy when writing incomplete data, and to continue
+    /// writing in chunks. Unlike the <see cref="Utf8JsonWriter"/>, which is a ref struct,
+    /// this type can survive across async/await boundaries and hence this type is required to provide
+    /// support for writing more JSON text asynchronously before continuing with a new instance of the <see cref="Utf8JsonWriter"/>.
+    /// </summary>
```

Comment:
If we're expecting these XML comments to become IntelliSense and docs, large summaries like this should be split into a short summary and a large remarks.